components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/dao/impl/FileBasedUserStoreDAOImpl.java

```diff
@@ -426,4 +436,72 @@ private UserStoreDTO getUserStoreDTO(RealmConfiguration secondaryRealmConfigurat
         }
         return userStoreDTO;
     }
+
+    /**
+     * To check whether the provided domain name is valid or not.
+     *
+     * @param domainName user store domain name
+     * @param isAdd      boolean value, this will be true when add a new user store
+     * @return true or false
+     * @throws IdentityUserStoreMgtException
+     */
+    private boolean isValidDomain(String domainName, Boolean isAdd) throws IdentityUserStoreMgtException {
+
+        if (StringUtils.isEmpty(domainName)) {
+            throw new IdentityUserStoreClientException(" User store domain name should not be empty.");
+        }
+
+        if (domainName.contains("_")) {
+            throw new IdentityUserStoreClientException(" User store domain name should not contain \"_\".");
+        }
+
+        if (isAdd) {
+            // if add, user store domain name shouldn't already exists
+            if (getDomainNames().contains(domainName)) {
```

Comment:
We can improve this logic to` if (isAdd && getDomainNames().contains(domainName))`